Automattic/wp-calypso
# Connect after checkout flow: Activate license after connection

This PR builds upon https://github.com/Automattic/wp-calypso/pull/82036 and redirects the user to the checkout thank-you license auto-activation page after connection takes place, allowing the user to activate their purchased product license key on their site they just purchased the product from.

Github task: https://github.com/orgs/Automattic/projects/724/views/1?pane=issue&itemId=40599568
Related to: https://github.com/Automattic/wp-calypso/pull/82036

## Proposed Changes

* Updated the `redirectAfterAuth` url to point to the checkout thank-you license auto activation page.
* Updated getThankYouPageUrl unit tests.
* Updated authorize.js to accept redirectAfterAuth to the checkout thank-you license auto activation page.
* Update license auto activation page to auto-select the site that we just purchased the product from.

## Testing Instructions

- Spin up a Jurassic.ninja site with the Jetpack Beta plugin running Jetpack branch `update/my-jetpack-connect-after-checkout`.
- Checkout this PR and `yarn start`. Make sure you are logged-in to WordPress.com.
- In the not-yet connected Jetpack plugin, go to: `/wp-admin/admin.php?page=my-jetpack&calypso_env=development`
- In My Jetpack, on the VaultPress Backup Card, click the "Purchase" Button.
- Click "Get Jetpack VaultPress Backup" button.
- Verify you are taken to checkout with Jetpack VaultPress Backup in the cart.
- On the checkout page, use A81N credits, and select "Assign a payment method later".
- Click "Complete checkout"
- Verify purchase is completed and you are redirected to the Connection/Authorization screen.
- Click "Approve".
- After connection, verify you are redirected to the Checkout Thank You Licensing Activation page (See screenshot), and that your site is already selected in the dropdown box.

#### Screenshot:

AFTER CONNECTION ⤵️
![Markup 2023-10-08 at 23 02 55](https://github.com/Automattic/wp-calypso/assets/11078128/68c9d106-7cf5-4329-8c17-562e1a02898b)

#### Screencast (Full checkout flow example)

https://github.com/Automattic/wp-calypso/assets/11078128/2a4fefdf-8e8e-454b-abf6-9d9f68c26e87




## Pre-merge Checklist
